File: example/src/main/java/com/quarkworks/android/realmtypesafequeryexample/TestRecord.java
package com.quarkworks.android.realmtypesafequeryexample;


import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.quarkworks.android.realmtypesafequery.annotations.GenerateRealmFieldNames;
import com.quarkworks.android.realmtypesafequery.annotations.GenerateRealmFields;

import java.util.Date;

import io.realm.RealmList;
import io.realm.RealmModel;
import io.realm.annotations.Ignore;
import io.realm.annotations.Index;
import io.realm.annotations.PrimaryKey;
import io.realm.annotations.RealmClass;

@RealmClass
@GenerateRealmFieldNames
@GenerateRealmFields
public class TestRecord implements RealmModel {

    @Nullable
    public Boolean booleanField;

    @Nullable
    public byte[] byteArrayField;

    @Nullable
    public Byte byteField;

    @Nullable
    public Date dateField;

    @Nullable
    public Double doubleField;

    @Nullable
    public Float floatField;

    @Nullable
    public Integer integerField;

    @Nullable
    public Long longField;

    @Nullable
    public Short shortField;

    @Nullable
    public String stringField;

    @Ignore @Nullable
    public Object ingnoredField;

    @Index
    public String indexedField;

    @PrimaryKey
    public String primaryKey;

    @Nullable
    public TestRecord parent = null;

    @NonNull
    public RealmList<TestRecord> children = new RealmList<>();
}

File: example_kotlinOnly/src/main/java/com/quarkworks/android/realmtypesafequery/example/TestRecord.kt
package com.quarkworks.android.realmtypesafequery.example


import com.quarkworks.android.realmtypesafequery.constants.RealmDefaults
import com.quarkworks.android.realmtypesafequery.annotations.GenerateRealmFieldNames
import com.quarkworks.android.realmtypesafequery.annotations.GenerateRealmFields

import java.util.Date

import io.realm.RealmList
import io.realm.RealmModel
import io.realm.annotations.Ignore
import io.realm.annotations.Index
import io.realm.annotations.PrimaryKey
import io.realm.annotations.RealmClass
import io.realm.annotations.Required

@RealmClass
@GenerateRealmFieldNames
@GenerateRealmFields
open class TestRecord : RealmModel {

    var booleanField: Boolean? = null

    internal var boolValue: Boolean = false

    var byteArrayField: ByteArray? = null

    var byteField: Byte? = null

    var byteValue: Byte = 0

    var dateField: Date? = null

    var doubleField: Double? = null

    internal var doubleValue: Double = 0.toDouble()

    var floatField: Float? = null

    internal var floatValue: Float = 0.toFloat()

    var integerField: Integer? = null

    var intValue: Int = 0

    var longField: Long? = null

    var longValue: Long = 0

    var shortField: Short? = null

    var shortValue: Short = 0

    var stringField: String? = null

    @Ignore
    var ignoredField: Object? = null

    @Index
    var indexedField: String? = null

    @PrimaryKey
    var primaryKey: String? = null

    @Required
    var requiredField = RealmDefaults.STRING

    var parent: TestRecord? = null

    var children: RealmList<TestRecord> = RealmList()
}
